pkg/operation/botanist/constraints_check.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+	"fmt"
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	gardencorev1alpha1helper "github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// ConstraintsChecker checks operational constraints of Shoot cluster.
+type ConstraintsChecker struct{}
```

Comment:
Why have you chosen to use a struct here although it doesn't hold any state inside?

In my opinion, it'd rather make sense if  `ConstrainChecker` implemented an interface, e.g.

```
type ConstraintChecker interface {
  CheckConstraint(constraint gardencorev1alpha1.Condition) gardencorev1alpha1.Condition
}
```

Reply:
As discussed, I simplified the constraints checks as follows: removed the `ConstraintsChecker` type and refactored the func `CheckWeebhook` to `IsProblematicWebhook` which now only checks a single webhook and the calling function loops over all Webbhooks in a `MutatingWebhookConfig`/`ValidatingWebhookConfig`.